### ISSUE TITLE
[Bug]: Dashboard chat sessions fail to render due to recent WebSocket regression

### ISSUE THREAD

**Issue by aarononeal** [NousResearch/hermes-agent#61284]
### Bug Description

Dashboard "Silent WebSocket" on Session Switch (v2026.7.7.2+)

## Symptom

When switching chat sessions in the Hermes dashboard, the PTY terminal opens but displays no output and the UI never updates. REST metadata fetching (`api.getSessionDetail`) succeeds. Browser console shows:

```
[chat] PTY WebSocket closed code=1005 reason=
ws closed peer=<ip> reason=client_disconnect(code=1005,reason=) messages=1 parse_errors=0 dispatch_crashes=0 send_failures=0 reaped_sessions=1 detached_sessions=0
```

The issue is that `PTY_REGISTRY.attach_or_spawn()` creates a new Node.js process that takes 1-3 seconds to boot, but the browser WebSocket has no mechanism to wait for output before considering the connection "established." The browser disconnects with `code=1005` before any PTY data arrives.

## Impact

- Session switching in the dashboard is non-functional
- Dashboard PTY shows blank terminal after session switch
- Active session (no switch) works fine — only the transition is broken
- The `ptyAttachToken(forceFresh=true)` path generates a new UUID on every session switch, creating a fresh PTY each time instead of reattaching

### Steps to Reproduce

1. Deploy Docker image v2026.7.7.2
2. Open dashboard, switch between 3+ chat sessions
3. All chat sessions will just re-render the same current session
4. Watch for `code=1005` `messages=1` in agent.log — should transition to `messages=N` (N > 1) after fix
5. Verify PTY renders output after each session switch within 5 seconds

### Expected Behavior

View should change to the selected chat session.

### Actual Behavior

View redraws the current or new chat session instead of the selected session.

### Affected Component

Other

### Debug Report

```shell
N/A - Fresh install of the docker image.
```

### Operating System

Dockerfile

### Additional Logs / Traceback (optional)

```shell

```

### Root Cause Analysis (optional)

**PTY Keep-Alive System Race Condition** — 6 new commits between v2026.7.1 and v2026.7.7.2 rewrote `/api/pty` from 1:1 socket↔PTY to registry-based with `?attach=<UUID>` token reattachment.

### The Race

| Step | v2026.7.1 (works) | v2026.7.7.2 (broken) |
|------|-------------------|---------------------|
| 1 | Session switch → old WS closes → PTY killed | `forceFresh=true` → new UUID token generated → old WS **detaches only** (PTY lives) |
| 2 | New WS opens → new PTY spawns immediately | `PTY_REGISTRY.attach_or_spawn()` creates new PTY process |
| 3 | Node.js boots, agent initializes (~1-3s) | Same — but browser disconnects **before** output arrives |
| 4 | Output flows to browser | Browser sees empty connection → `code=1005`, `messages=1` |

### Why `messages=1` + `code=1005`

The browser WebSocket opens (1 frame = the handshake), then closes before any PTY output arrives. The PTY process exists but hasn't produced output yet — Node.js is still booting, the agent hasn't started, the drain task hasn't begun forwarding.

## Code Changes (v2026.7.1 → v2026.7.7.2)

| Commit | File | Change |
|--------|------|--------|
| `41166bbe` | `hermes_cli/pty_session.py` | `PtySessionRegistry` with capacity + reap |
| `0ecfbc98` | `hermes_cli/pty_session.py` | `RingBuffer` for output replay |
| `e5ac169c` | `hermes_cli/pty_session.py` | `.attach()`/`.detach()` lifecycle, close codes 4409/4410 |
| `e10e4bca` | `hermes_cli/web_server.py` | `?attach=<token>` reattach path (139 additions, 59 deletions) |
| `79f4f78f` | `web/src/pages/ChatPage.tsx` | `ptyAttachToken()` localStorage + auto-reconnect (55 additions) |
| `c3d2be07` | `hermes_cli/web_server.py` | Periodic reaper wired into dashboard lifespan |

@TinkerOfThings 

### Proposed Fix (optional)

1. **Add a ready-check before closing** — wait for the drain task to start forwarding before allowing the browser WS to timeout
2. **Use `ptyAttachToken(forceFresh=false)` during session switch** — keep the same token so the registry reuses the existing PTY instead of spawning a new one
3. **Increase the browser reconnect backoff** — give Node.js time to boot before the browser gives up and disconnects
4. **Add a `/api/pty/ready` endpoint** — frontend polls this before considering the connection "established"

### Are you willing to submit a PR for this?

- [ ] I'd like to fix this myself and submit a PR

**Comment by jbienz** [comment]
I believe this should be a higher priority since it blocks resuming of all previous sessions in the web dashboard. Luckily the Desktop app appears unaffected, but this is still high impact.

**Comment by TinkerOfThings** [comment]
The fix for this is **#60745** (attach-token session-scoping) — that PR keys the PTY keep-alive registry on `attach\0profile\0resume` instead of `attach` alone, so switching to `?resume=B` spawns a fresh PTY instead of reattaching to the still-live session-A pty. That's the actual root cause here; the `code=1005 / messages=1` teardown in the report is a benign superseded-socket close, not the bug.

I've left a small non-blocking hardening suggestion on that PR to reject a literal NUL (`%00`) in the keyed params before they're folded into the `\0`-joined key (otherwise `attach='a\0b'` could alias `attach='a'`+`resume='b'`): https://github.com/NousResearch/hermes-agent/pull/60745#discussion_r3572966133

**Comment by konsisumer** [comment]
These issues look like facets of **one underlying problem**: #61284, #63701, #64017.

They were grouped by a semantic pass as different symptoms of one underlying problem in `comp/dashboard`. A single coordinated fix may resolve them together — flagging for visibility; no action has been taken on the individual issues.

Related, already covered by an earlier root-cause note: #53839 (see the root-cause note on #53839).

_Automated root-cause grouping. Reply to correct if these are unrelated._
